internal/resources/providers/awslib/config.go

```diff
@@ -18,28 +18,102 @@
 package awslib
 
 import (
+	"context"
+	"fmt"
 	"net/http"
+	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
 	libbeataws "github.com/elastic/beats/v7/x-pack/libbeat/common/aws"
+
+	"github.com/elastic/cloudbeat/internal/config"
 )
 
+func awsConfigRetrier() aws.Retryer {
+	return retry.NewStandard(func(o *retry.StandardOptions) {
+		o.Retryables = append(o.Retryables, retry.RetryableHTTPStatusCode{
+			Codes: map[int]struct{}{
+				http.StatusTooManyRequests: {},
+			},
+		})
+	})
+}
+
 func InitializeAWSConfig(cfg libbeataws.ConfigAWS) (*aws.Config, error) {
 	awsConfig, err := libbeataws.InitializeAWSConfig(cfg)
 	if err != nil {
 		return nil, err
 	}
 
-	awsConfig.Retryer = func() aws.Retryer {
-		return retry.NewStandard(func(o *retry.StandardOptions) {
-			o.Retryables = append(o.Retryables, retry.RetryableHTTPStatusCode{
-				Codes: map[int]struct{}{
-					http.StatusTooManyRequests: {},
-				},
-			})
-		})
-	}
+	awsConfig.Retryer = awsConfigRetrier
 
 	return &awsConfig, nil
 }
+
+func CloudConnectorsExternalID(resourceID, externalIDPart string) string {
+	return fmt.Sprintf("%s-%s", resourceID, externalIDPart)
+}
+
+func InitializeAWSConfigCloudConnectors(ctx context.Context, cfg config.AwsConfig) (*aws.Config, error) {
+	// 1. Load initial config
+	// (TODO: check directly assuming the first role in chain and/or libbeataws.InitializeAWSConfig(cfg))
+	// (TODO: consider os.Setenv("AWS_EC2_METADATA_DISABLED", "true"))
```

Comment:
I am not sure what to consider, to set it or to check the behavior when it was previously set?
Also, in MKI we block the IMDS ip address, how it affect the flow?

Reply:
The comment was for me (I will remove them) to check id we need to use the AWS_EC2_METADATA_DISABLED in case we needed to remove the firewall rules. We don't (we checked the flow with @amirbenun a while ago) so all good. 

No changes needed. I will remove both TODOs.